{"ast":null,"code":"import _classCallCheck from \"C:/pproject-files/new-app/node_modules/@babel/runtime/helpers/esm/classCallCheck.js\";\nimport _createClass from \"C:/pproject-files/new-app/node_modules/@babel/runtime/helpers/esm/createClass.js\";\nimport _inherits from \"C:/pproject-files/new-app/node_modules/@babel/runtime/helpers/esm/inherits.js\";\nimport _createSuper from \"C:/pproject-files/new-app/node_modules/@babel/runtime/helpers/esm/createSuper.js\";\nimport * as React from 'react';\nexport var ErrorBoundary = /*#__PURE__*/function (_React$Component) {\n  _inherits(ErrorBoundary, _React$Component);\n\n  var _super = _createSuper(ErrorBoundary);\n\n  function ErrorBoundary() {\n    _classCallCheck(this, ErrorBoundary);\n\n    return _super.apply(this, arguments);\n  }\n\n  _createClass(ErrorBoundary, [{\n    key: \"componentDidCatch\",\n    value: function componentDidCatch(error, errorInfo) {\n      if (this.props.api.current) {\n        this.logError(error); // Allows to trigger the Error event and all listener can run on Error\n\n        this.props.api.current.showError({\n          error: error,\n          errorInfo: errorInfo\n        });\n      }\n    }\n  }, {\n    key: \"logError\",\n    value: function logError(error, errorInfo) {\n      this.props.logger.error(\"An unexpected error occurred. Error: \".concat(error && error.message, \". \"), error, errorInfo);\n    }\n  }, {\n    key: \"render\",\n    value: function render() {\n      var _this$state;\n\n      if (this.props.hasError || (_this$state = this.state) != null && _this$state.hasError) {\n        // You can render any custom fallback UI\n        return this.props.render(this.props.componentProps || this.state);\n      }\n\n      return this.props.children;\n    }\n  }], [{\n    key: \"getDerivedStateFromError\",\n    value: function getDerivedStateFromError(error) {\n      // Update state so the next render will show the fallback UI.\n      return {\n        hasError: true,\n        error: error\n      };\n    }\n  }]);\n\n  return ErrorBoundary;\n}(React.Component);","map":{"version":3,"sources":["C:/pproject-files/new-app/node_modules/@mui/x-data-grid/components/ErrorBoundary.js"],"names":["React","ErrorBoundary","error","errorInfo","props","api","current","logError","showError","logger","message","_this$state","hasError","state","render","componentProps","children","Component"],"mappings":";;;;AAAA,OAAO,KAAKA,KAAZ,MAAuB,OAAvB;AACA,WAAaC,aAAb;AAAA;;AAAA;;AAAA;AAAA;;AAAA;AAAA;;AAAA;AAAA;AAAA,WASE,2BAAkBC,KAAlB,EAAyBC,SAAzB,EAAoC;AAClC,UAAI,KAAKC,KAAL,CAAWC,GAAX,CAAeC,OAAnB,EAA4B;AAC1B,aAAKC,QAAL,CAAcL,KAAd,EAD0B,CACJ;;AAEtB,aAAKE,KAAL,CAAWC,GAAX,CAAeC,OAAf,CAAuBE,SAAvB,CAAiC;AAC/BN,UAAAA,KAAK,EAALA,KAD+B;AAE/BC,UAAAA,SAAS,EAATA;AAF+B,SAAjC;AAID;AACF;AAlBH;AAAA;AAAA,WAoBE,kBAASD,KAAT,EAAgBC,SAAhB,EAA2B;AACzB,WAAKC,KAAL,CAAWK,MAAX,CAAkBP,KAAlB,gDAAgEA,KAAK,IAAIA,KAAK,CAACQ,OAA/E,SAA4FR,KAA5F,EAAmGC,SAAnG;AACD;AAtBH;AAAA;AAAA,WAwBE,kBAAS;AACP,UAAIQ,WAAJ;;AAEA,UAAI,KAAKP,KAAL,CAAWQ,QAAX,IAAuB,CAACD,WAAW,GAAG,KAAKE,KAApB,KAA8B,IAA9B,IAAsCF,WAAW,CAACC,QAA7E,EAAuF;AACrF;AACA,eAAO,KAAKR,KAAL,CAAWU,MAAX,CAAkB,KAAKV,KAAL,CAAWW,cAAX,IAA6B,KAAKF,KAApD,CAAP;AACD;;AAED,aAAO,KAAKT,KAAL,CAAWY,QAAlB;AACD;AAjCH;AAAA;AAAA,WACE,kCAAgCd,KAAhC,EAAuC;AACrC;AACA,aAAO;AACLU,QAAAA,QAAQ,EAAE,IADL;AAELV,QAAAA,KAAK,EAALA;AAFK,OAAP;AAID;AAPH;;AAAA;AAAA,EAAmCF,KAAK,CAACiB,SAAzC","sourcesContent":["import * as React from 'react';\nexport class ErrorBoundary extends React.Component {\n  static getDerivedStateFromError(error) {\n    // Update state so the next render will show the fallback UI.\n    return {\n      hasError: true,\n      error\n    };\n  }\n\n  componentDidCatch(error, errorInfo) {\n    if (this.props.api.current) {\n      this.logError(error); // Allows to trigger the Error event and all listener can run on Error\n\n      this.props.api.current.showError({\n        error,\n        errorInfo\n      });\n    }\n  }\n\n  logError(error, errorInfo) {\n    this.props.logger.error(`An unexpected error occurred. Error: ${error && error.message}. `, error, errorInfo);\n  }\n\n  render() {\n    var _this$state;\n\n    if (this.props.hasError || (_this$state = this.state) != null && _this$state.hasError) {\n      // You can render any custom fallback UI\n      return this.props.render(this.props.componentProps || this.state);\n    }\n\n    return this.props.children;\n  }\n\n}"]},"metadata":{},"sourceType":"module"}